Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -0,0 +1,1307 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+
+type OutOfBandRequestSuccessHandler = (args: { Status: number}) => void;
+type OutOfBandRequestErrorHandler = (args: { Status: number}) => void;
+
+type SubmitState = {
+  CancelSubmit: boolean;
+  IsAsynchronous: boolean;
+  IsAutoPostback: boolean;
+  Submitter: Nullable<HTMLElement>;
+  EventTarget: string;
+  EventArgument: string;
+  NextSubmitState: Nullable<SubmitState>;
+};
+
+type NextSubmitState = {
+  CancelSubmit: boolean;
+  IsAsynchronous?: boolean;
+  IsAutoPostback?: boolean;
+  Submitter: Nullable<HTMLElement>;
+  EventTarget?: string;
+  EventArgument?: string;
+  NextSubmitState?: Nullable<NextSubmitState>;
+};
+
+type SmartPageCheckFormState = (_isAborting: boolean, _hasSubmitted: boolean, _hasUnloaded: boolean, _isCached: boolean) => boolean;
+
+type SmartPageOnLoadingEventHandler = (this: null, hasSubmitted: boolean, isCached: boolean, isAsynchronous: boolean) => void;
```

Comment:
Because it is always called with `null` as `this` parameter:

https://github.com/re-motion/Framework/blob/3a6c3497f45ffa73b1d6972f430fa9ea3d1159f5/Remotion/Web/ClientScript/Scripts/SmartPage.ts#L874

Reply:
i.c.